workshops/payments/render.py

```diff
@@ -0,0 +1,37 @@
+import os
+from io import BytesIO
+from django.template.loader import get_template
+import xhtml2pdf.pisa as pisa
+
+
+def render(path: str, params: dict):
+    template = get_template(path)
+    html = template.render(params)
+    response = BytesIO()
+    pisa.pisaDocument(BytesIO(html.encode("UTF-8")), response, link_callback=fetch_resources)
+    return response.getvalue()
+
+
+# This was taken from the xhtml2pdf documentation then modified because they
+# didn't account for ../ in relative filepaths and sRoot wasn't correct
+def fetch_resources(uri, rel):
+    """
+    Convert HTML URIs to absolute system paths so xhtml2pdf can access those
+    resources
+    """
+    uri = uri.replace('../', '')
+    # use short variable names
+    sUrl = 'static/'
```

Comment:
`settings.STATIC_URL` is set to '/static/', but I believe I need it to be 'static/' to work. The image uri in invoice.html is '../static/img/q2logo.png' on line 22 I replace the '../' with a '' so I can join it with '../workshops/workshops/' on line 35 to get the full path. I attempted only replacing the '..' with a '' and joining the resulting '/static/img/q2logo.png' with both '../workshops/workshops/' and '../workshops/workshops' and neither result in the paths actually joining properly. I verified in ipython that `os.path.join` does not properly join those paths after I attempted both and didn't see the image in the result.

Reply:
So I suppose the question now becomes.
1. Am I doing something wrong preventing me from using '/static/'?
2. Do I change `settings.STATIC_URL` to 'static/' instead? (at a glance, I'm not seeing it used anywhere else)
3. Do I leave things as they are?

Reply:
Please take a closer look at the [xhtml2pdf docs](https://xhtml2pdf.readthedocs.io/en/latest/usage.html#using-xhtml2pdf-in-django), the boilerplate there appears to have `STATIC_URL='/static/'`, which leads me to believe it might be that it is a simple fix needed on your end.